### PR TITLE
Update Android to R21e

### PR DESCRIPTION
**Pull request by cyyever**
R19c is too old, R21e is LTS version.

### CODE REVIEW

File: .ci/docker/common/install_android.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ apt-get autoclean && apt-get clean
 rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
 pushd /tmp
-curl -Os --retry 3 $_https_amazon_aws/android-ndk-${ANDROID_NDK}-linux-x86_64.zip
+curl -Os --retry 3 https://dl.google.com/android/repository/android-ndk-${ANDROID_NDK}-linux-x86_64.zip
```
Comment (malfet) [inline .ci/docker/common/install_android.sh:15]: Please don't do that, as `dl.google.com` often rate-limits our CI. Let me upload NDK-19 to our mirror

**Review by malfet — CHANGES_REQUESTED** [review]
Please don't change the download URL (uploading NDK to our mirror)

File: .ci/docker/common/install_android.sh (hunk 1 above)
Comment (malfet) [reply .ci/docker/common/install_android.sh:15]: Done

Comment (cyyever) [reply .ci/docker/common/install_android.sh]: Fixed, but I need your help to upload.

**Review by malfet — APPROVED** [review]
LGTM if CI is green. Thank you for the updates.
[Edit] Will debug what is going on with the downloads

**Review by huydhn — APPROVED** [review]
IIRC, I encounter an build issue with Vulkan when trying this locally.  Let's see if this pass CI

**Comment by cyyever** [comment]
Thank you

**Comment by huydhn** [comment]
Btw, you might also want to rebase this to include https://github.com/pytorch/pytorch/pull/110976.  I just land this to add a new release workflow for Android.

**Comment by cyyever** [comment]
@huydhn wait some hours until it goes into viable/strict

**Comment by cyyever** [comment]
@huydhn I have no permission to docker push, help me verify the PR?

**Comment by huydhn** [comment]
> @huydhn I have no permission to docker push, help me verify the PR?

Sorry, I missed your message. I will need to prepare the ECR entry for the new docker images as the names are different.  After that, rerunning the job will work

**Comment by cyyever** [comment]
@huydhn Is it able to merge?

**Comment by cyyever** [comment]
@huydhn  Now there is an error
···
ANDROID_HOME not a directory; did you install it under /opt/android/sdk?
···

**Comment by huydhn** [comment]
Hmm, I don't see any issue in trunk.  Is that failing locally for you?  The SDK should be under that path https://github.com/pytorch/pytorch/blob/main/.ci/docker/common/install_android.sh#L51

**Pull request closed without merging**